storage/remote/queue_manager.go

```diff
@@ -1116,19 +1116,33 @@ func (q *queue) ReturnForReuse(batch []sampleOrExemplar) {
 // FlushAndShutdown stops the queue and flushes any samples. No appends can be
 // made after this is called.
 func (q *queue) FlushAndShutdown(done <-chan struct{}) {
+	for q.trySendingBatch(done) {
+		time.Sleep(time.Second)
+	}
+	q.batch = nil
+	close(q.batchQueue)
+}
+
+// trySendingBatch tries to send a batch if necessary. If sending needs to be
+// retried it will return true.
+func (q *queue) trySendingBatch(done <-chan struct{}) bool {
```

Comment:
can we rename this `tryEnqueuingBatch`? `Sending` initially makes me think we're doing the remote write from this function

Reply:
:+1: